### ISSUE TITLE
How to get event on onSubmit handler?

### ISSUE THREAD

**Issue by tero** [davidkpiano/react-redux-form#637]
In my form logic I have to know if form is submitted from keyboard or by clicking submit button, but can't access this information because event isn't passed to onSubmit handler. Is there any workaround how to get the original event?

**Comment by davidkpiano** [comment]
I'll admit, this was sort of an API design mistake which will be addressed in V2 (see #626), but it's one that can easily be patched for now. It's just a question of what the (temporary) API should be:

- `<Form onSubmitEvent={(e) => ...}>`
- `<Form onBeforeSubmit={(e, values) => ...}>`
- Any other suggestions?

In V2, all event handlers will behave the same as normal event handlers, where the `event` is passed in as the first argument to the handler. Currently, `<Control>` has a `withField={true}` prop that passes in the `fieldValue` as the second argument. I'm envisioning the same for `<Form>` in the future:

```jsx
const handleSubmit = (event, formValue) => { ... };

<Form model="user" onSubmit={handleSubmit}>
```

**Comment by tero** [comment]
Have you scheduled any roadmap yet? When v2 will be out and what kind of changes is planned there? 

For temporary API my vote goes for onBeforeSubmit.
